pkg/spiffe/spiffe.go

```diff
@@ -90,3 +102,60 @@ func GenCustomSpiffe(identity string) string {
 
 	return URIPrefix + GetTrustDomain() + "/" + identity
 }
+
+// RetrieveSpiffeBundleRootCert retrieves the trusted CA certificate from a SPIFFE bundle endpoint.
+// It can use the system cert pool and the supplied certificates to validate the endpoint.
+func RetrieveSpiffeBundleRootCert(endpoint string, extraTrustedCerts []*x509.Certificate) (*x509.Certificate, error) {
```

Comment:
1. Please check the content type of the result - pretty sure the spiffee or JWK define a content-type for the response
2. I'm pretty sure we'll also want to support getting the roots as PEM - for non-spiffee CAs we may support. So 
maybe break this method in 2, one part that handles the response ( after checking content type), and a method that is fetching the response. The second method may move and change to also support retrieving PEM roots, and also could use files in future ( i.e. configmaps).